templates/server/puppetserver/conf.d/ca.conf.erb

```diff
@@ -0,0 +1,22 @@
+# CA-related settings
+certificate-authority: {
+
+    # settings for the certificate_status HTTP endpoint
+    certificate-status: {
+
+        # this setting contains a list of client certnames who are whitelisted to
+        # have access to the certificate_status endpoint. Any requests made to
+        # this endpoint that do not present a valid client cert mentioned in
+        # this list will be denied access.
+<%- if @server_ca_auth_required == false -%>
+        authorization-required: false
+<%- else -%>
+        authorization-required: true
+        client-whitelist: [
+    <%- @server_ca_client_whitelist.each do |client| -%>
+            <%= client %>,
```

Comment:
Not 100% sure if it has any negative consequences on IPv4 addresses, but at least IPv6 addresses need to get double-quoted here, otherwise `::1` will result into: `Exception in thread "main" com.typesafe.config.ConfigException$Parse: /etc/puppetlabs/puppetserver/conf.d/ca.conf: 14: in value for key 'client-whitelist': List should have ended with ] or had a comma, instead had token: ':' (if you want ':' to be part of a string value, then double-quote it)`